src/ripple/basics/SlabAllocator.h

```diff
@@ -57,40 +57,66 @@ class SlabAllocator
         // A linked list of appropriately sized free buffers:
         std::uint8_t* l_ = nullptr;
 
-        // The next memory block
+        // The next block:
         SlabBlock* next_;
 
         // The underlying memory block:
-        std::uint8_t const* const p_ = nullptr;
+        std::atomic<std::uint8_t*> data_ = nullptr;
```

Comment:
With one exception, `data_` is always access within a lock. I _think_ we can get rid of that exception (see comment below); if we do I don't think this needs to be `atomic`